### PR TITLE
Created controller for fetch items

### PR DESCRIPTION
**Pull request by VladyslavAnokhin**
## Changes

- Created fetch item controller
- Refactored VRMItem

[JIRA Ticket](https://jira.ouroath.com/browse/OMSDK-2014)

@OathAdPlatforms/mobile-sdk-developers: Please review.

### CODE REVIEW

**Comment by VladyslavAnokhin** [comment]
@AndriiMoskvin RFR

File: sources/OVPSDK.swift [hunk 1]
```diff
@@ -406,23 +406,27 @@ public struct OVPSDK {
         }
         
         func setupVRMWithNewCore() {
+            let createRequest: (URL) -> (URLRequest) = {
```
Comment (AndriiMoskvin) [inline sources/OVPSDK.swift:409]: This have to be placed closer to the actual call place.

File: sources/OVPSDK.swift [hunk 2]
```diff
@@ -406,23 +406,27 @@ public struct OVPSDK {
...
+                .init(url: $0, cachePolicy: .useProtocolCachePolicy, timeoutInterval: hardTimeout)
```
Comment (AndriiMoskvin) [inline sources/OVPSDK.swift]: .useProtocolCachePolicy - is a default value?

File: sources/advertisements/VRM New Core/Controllers/FetchVRMItemController.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import PlayerCore
+
+final class FetchVRMItemController {
+    
+    typealias VASTXMLFetcher = (URL) -> Future<Result<String>>
+
+    let dispatch: (PlayerCore.Action) -> Void
+    let fetchUrl: VASTXMLFetcher
+    
+    private var startedItems = Set<PlayerCore.VRMFetchItemQueue.FetchCandidate>()
```
Comment (AndriiMoskvin) [inline sources/advertisements/VRM New Core/Controllers/FetchVRMItemController.swift]: `PlayerCore.VRMFetchItemQueue.Candidate`

File: sources/advertisements/VRM New Core/Controllers/FetchVRMItemController.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
+//  Copyright 2018, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+
+import Foundation
+import PlayerCore
+
+final class FetchVRMItemController {
+    
+    typealias VASTXMLFetcher = (URL) -> Future<Result<String>>
```
Comment (AndriiMoskvin) [inline sources/advertisements/VRM New Core/Controllers/FetchVRMItemController.swift]: do we need this typealias - it makes me confused - the signature is pretty self explanatory

File: sources/advertisements/VRM New Core/Controllers/FetchVRMItemControllerTest.swift [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+@testable import OathVideoPartnerSDK
+@testable import PlayerCore
+
+class FetchVRMItemControllerTest: XCTestCase {
+    
+    let recorder = Recorder()
+    let parseActionComparator = ActionComparator<VRMCore.StartItemParsing> {
+        $0.vastXML == $1.vastXML && $0.originalItem == $1.originalItem
+    }
+    let faildFetchActionCompare = ActionComparator<VRMCore.FetchingError> {
```
Comment (AndriiMoskvin) [inline sources/advertisements/VRM New Core/Controllers/FetchVRMItemControllerTest.swift]: `failed` Why not use Equatable?

**Review by AndriiMoskvin — DISMISSED** [review]

File: sources/OVPSDK.swift (hunk 2 above)
Comment (VladyslavAnokhin) [reply sources/OVPSDK.swift]: dropped

File: sources/advertisements/VRM New Core/Controllers/FetchVRMItemController.swift (hunk 4 above)
Comment (VladyslavAnokhin) [reply sources/advertisements/VRM New Core/Controllers/FetchVRMItemController.swift]: dropped

File: sources/advertisements/VRM New Core/Controllers/FetchVRMItemController.swift (hunk 3 above)
Comment (VladyslavAnokhin) [reply sources/advertisements/VRM New Core/Controllers/FetchVRMItemController.swift]: Renamed

File: sources/OVPSDK.swift (hunk 1 above)
Comment (AndriiMoskvin) [reply sources/OVPSDK.swift:409]: Still prefer to avoid such function and write them inline whenever possible.

**Review by AndriiMoskvin — APPROVED** [review]

File: sources/OVPSDK.swift (hunk 1 above)
Comment (VladyslavAnokhin) [reply sources/OVPSDK.swift:409]: In future I'm going to replace `hardTimeout` with other value which comes from MS. So such structure helps not forget to change this value in all places where I've created URLRequest

Comment (AndriiMoskvin) [reply sources/OVPSDK.swift:409]: If I correctly understand there will be ONLY ONE such place.

Comment (VladyslavAnokhin) [reply sources/OVPSDK.swift:409]: There are two places already

**Pull request merged**